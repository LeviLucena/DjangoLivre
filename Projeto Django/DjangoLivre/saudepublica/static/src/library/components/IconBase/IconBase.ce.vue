<template>
  <i :class="iconClasses" aria-hidden="true"></i>
</template>

<script>
const IconBase = {
  name: "IconBase",
  props: {
    /**
     * Nome do ícone que será apresentado
     */
    iconName: {
      type: String,
      default: "",
      validator: function (val) {
        return val.trim().length > 1
      },
    },
    /**
     * Se informado define que o nome da familia de fonte que será aplicada ao ícone
     */
    familyName: {
      type: String,
      default: "fas",
      validator: function (val) {
        return val.trim().length > 1
      },
    },
    /**
     * Se informado define qual o estilo de classe css que será aplicada ao ícone
     */
    className: {
      type: [String, Object],
      default: null,
    },
  },
  computed: {
    /**
     * Retorna o nome do ícone que será aplicado ao componente
     */
    iconClasses() {
      return [this.familyName, `fa-${this.processIconName(this.iconName)}`, this.className].join(" ")
    },
  },
  methods: {
    /**
     * Retorna o nome processado do ícone que será
     * apresentado
     */
    processIconName(iconName) {
      // Remove o prefixo "fa-" se estiver presente
      return iconName.replace(/^fa-/, "")
    },
  },
}

export default IconBase
</script>

<style lang="scss">
@import "~@govbr-ds/core/src/partial/scss/_base";
@import "~@govbr-ds/core/src/partial/scss/_utilities";
@import url("https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/all.min.css");
</style>
