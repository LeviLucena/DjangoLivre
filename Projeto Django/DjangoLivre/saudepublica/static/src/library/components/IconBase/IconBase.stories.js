import { kebabiseArgs } from '../../util/Utils.js'
import IconBase from './IconBase.ce.vue'

export default {
  title: 'Componentes/Icon',
  component: IconBase,
  argTypes: {
    iconName: {
      control: {
        type: 'text',
      },
    },
    familyName: {
      control: {
        type: 'text',
      },
    },
    className: {
      control: {
        type: 'text',
      },
    },
  },
}

const Template = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<br-icon-base v-bind="args"></br-icon-base>`,
})

export const Padrão = Template.bind({})
Padrão.args = {
  iconName: 'home',
  familyName: 'fas',
  className: '',
}
