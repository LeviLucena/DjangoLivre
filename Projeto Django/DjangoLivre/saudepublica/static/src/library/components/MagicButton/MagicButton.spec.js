import { shallowMount } from '@vue/test-utils'
import BrMagicButton from './MagicButton.ce.vue'
import IconBase from '../IconBase/IconBase.ce.vue'

const densidades = ['large', 'medium', 'small']

describe('MagicButton', () => {
  test('it renders br-magic-button', () => {
    const rotulo = 'Label'
    const wrapper = shallowMount(BrMagicButton, {
      propsData: {
        label: rotulo,
      },
    })
    expect(wrapper.classes('br-magic-button')).toBe(true)
  })

  test('set label attribute to create pill type', () => {
    const rotulo = 'Label'
    const wrapper = shallowMount(BrMagicButton, {
      propsData: {
        label: rotulo,
      },
    })
    expect(wrapper.text()).toMatch(rotulo)
  })

  test('set icon', () => {
    const wrapper = shallowMount(BrMagicButton, {
      propsData: {
        icon: 'car',
        circle: true,
      },
      global: {
        stubs: {
          'icon-base': IconBase,
        },
      },
    })
    expect(wrapper.props().icon).toBe('car')
    expect(wrapper.element).toMatchSnapshot()
  })

  densidades.forEach((densidade) => {
    test(`set density attribute ${densidade}`, () => {
      const rotulo = 'Texto do Bot√£o'
      const wrapper = shallowMount(BrMagicButton, {
        propsData: {
          density: densidade,
          label: rotulo,
        },
      })
      expect(wrapper.find(`.br-magic-button.${densidade}`).exists()).toBe(true)
    })
  })
})
