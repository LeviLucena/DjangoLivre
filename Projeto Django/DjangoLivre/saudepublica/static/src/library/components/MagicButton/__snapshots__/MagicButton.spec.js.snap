// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MagicButton set icon 1`] = `
<div
  class="br-magic-button medium"
>
  <button
    class="br-button circle"
    type="button"
  >
    <i
      aria-hidden="true"
      class="fas fa-car"
    />
     
  </button>
</div>
`;
